lib/rules/no-window-top.js

```diff
@@ -22,7 +22,13 @@ module.exports = {
     create(context) {
         return {
             'Program:exit'() {
-                const ref = context.getScope().through.find(({ identifier }) => {
+                // The top level scope is the "global" scope. Depending on the "env" property in the
+                // eslint config, references might or not be resolved at the global scope. The most
+                // reliable way to get all the unresolved references from the module is to get them
+                // from the "module" scope.
+                const moduleScope = context.getScope().childScopes[0];
```

Comment:
Same here, IMO its better to explicitly find the module scope instead of implicitly getting the first childScope